### PR TITLE
Make dynamic macro feature easier to enable and use

### PR DESCRIPTION
**Pull request by snyman**
Now it works like other features, without weird import order dependencies or extra code required in the keymap files (although that is still supported for more advanced use cases through an extra make variable).

### CODE REVIEW

**Comment by drashna** [comment]
There are a bunch of merge conflicts here.

Would you mind updating this?

**Comment by yanfali** [comment]
Now this is a .c file and not a .h file how much memory does it use?

File: keyboards/hhkb/keymaps/mjt/keymap.c [hunk 1]
```diff
@@ -2,6 +2,7 @@
  * default HHKB Layout
  */
 #include QMK_KEYBOARD_H
+#include "dynamic_macro.h"
```
Comment (skullydazed) [inline keyboards/hhkb/keymaps/mjt/keymap.c]: I'm not a big fan of changing user keymaps. Will these continue to work without being changed, or will this PR cause them to start erroring out?

**Comment by drashna** [comment]
@snyman there are a bunch of merge conflicts here, that need to be fixed. 

@skullydazed I think for this, the keymap changes are necessary, since it's a big change to how the feature works. 

Maybe "breaking changes" is appropriate here?

**Comment by fauxpark** [comment]
I think dynamic_macro.[ch] could be moved under process_keycode/ as process_dynamic_macro.[ch], with `process_record_dynamic_macro()` renamed to `process_dynamic_macro()`.

**Comment by drashna** [comment]
@fauxpark Yeah, that would probably be best, as it keeps things consistent.

**Comment by drashna** [comment]
@snyman any updates on this?

**Comment by drashna** [comment]
I'm closing this PR.  #5948 is based directly on this, and is updated, as well.

**Pull request closed without merging**